theano/gpuarray/subtensor.py

```diff
@@ -1108,15 +1107,19 @@ def gpu_kernels(self, node, nodename):
   old = *base;
   do {
     assumed = old;
-    sum = __float2half_rn(
+    ga_half old_perm;
+    __HALF_TO_US(old_perm)  = __byte_perm(old, 0,
+                     ((ga_size)addr & 2) ? 0x4432 : 0x4410);
+    sum = __float2half_as_us(
       __half2float(val) +
-      __half2float((ga_half)__byte_perm(old, 0,
-                     ((ga_size)addr & 2) ? 0x4432 : 0x4410)));
+      __half2float(old_perm));
     new_ = __byte_perm(old, sum, ((ga_size)addr & 2) ? 0x5410 : 0x3254);
     old = atomicCAS(base, assumed, new_);
   } while (assumed != old);
-  return (ga_half)__byte_perm(old, 0,
-                                  ((ga_size)addr & 2) ? 0x4432 : 0x4410);
+  ga_half ret;
+  __HALF_TO_US(ret) = __byte_perm(old, 0,
+        ((ga_size)addr & 2) ? 0x4432 : 0x4410);
+  return ret;
```

Comment:
I want to move those definitions to libgpuarray so that we don't need to provide them manually everywhere.